### PR TITLE
Convert react File Manager and HtmlEditor demos to functionality

### PR DESCRIPTION
**Pull request by dxvladislavvolkov**

### CODE REVIEW

File: JSDemos/Demos/FileManager/AzureClientBinding/React/App.js [hunk 1]
```diff
@@ -172,7 +75,90 @@ function downloadItems(items) {
   azure.downloadFile(items[0].path);
 }
 
-let gateway = null;
-let azure = null;
+function App() {
+  gateway = new AzureGateway(endpointUrl, onRequestExecuted);
```
Comment (ivanblinov2k17) [inline JSDemos/Demos/FileManager/AzureClientBinding/React/App.js]: seems like on every state change there will be new gateway. think that can be moved to useeffect

File: JSDemos/Demos/FileManager/UICustomization/React/App.js [hunk 2]
```diff
@@ -171,15 +103,15 @@ class App extends React.Component {
...
 
-  updateCategory(newCategory, directory, viewArea) {
+  const updateCategory = React.useCallback((newCategory, directory, viewArea) => {
     let items = null;
 
     if (viewArea === 'navPane') {
       items = [directory];
     } else {
-      items = this.fileManager.getSelectedItems();
+      items = getFileManagerInstance.getSelectedItems();
```
Comment (ivanblinov2k17) [inline JSDemos/Demos/FileManager/UICustomization/React/App.js]: missing function call?

Comment (ivanblinov2k17) [reply JSDemos/Demos/FileManager/UICustomization/React/App.js]: maybe remove this function and leave fileManagerRef.current.instance as it is?

**Review by alexanderPolosatov — COMMENTED** [review]
LGTM

**Review by ivanblinov2k17 — APPROVED** [review]

**Pull request merged**